Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/serverexplore/AddHDInsightAdditionalClusterImpl.java

```diff
@@ -48,12 +50,16 @@
 
 public class AddHDInsightAdditionalClusterImpl {
 
-    private static final String clusterConfigureFileUrl = "https://%s.azurehdinsight.net/api/v1/clusters/%s/configurations/service_config_versions?service_name=HDFS&service_config_version=1";
+    // private static final String clusterConfigureFileUrl = "https://%s.azurehdinsight.net/api/v1/clusters/%s/configurations/service_config_versions?service_name=HDFS&service_config_version=1";
 
     private static final Pattern PATTERN_DEFAULT_STORAGE = Pattern.compile("\"fs\\.defaultFS\":\"wasb://([^@\"]*)@([^@\"]*)\"", Pattern.CASE_INSENSITIVE | Pattern.COMMENTS);
     private static final Pattern PATTER_STORAGE_KEY = Pattern.compile("\"fs\\.azure\\.account\\.key\\.[^\"]*\":\"[^\"]*=\"", Pattern.CASE_INSENSITIVE | Pattern.COMMENTS);
     private static final String STORAGE_ACCOUNT_NAME_PATTERN = "^wasb://(.*)@(.*)$";
-    private static final String BLOB_URL_SUFFIX = ".blob.core.windows.net";
+
+    private static final String getClusterConfigureFileUrl(@NotNull final String clusterName) {
+        final String url = ClusterManagerEx.getInstance().getClusterConnectionString(clusterName) + "api/v1/clusters/%s/configurations/service_config_versions?service_name=HDFS&service_config_version=1";
```

Comment:
Any other solution to avoid this hardcode?

Reply:
For function getClusterConnectionString, I just rewrite it. Thanks.

Reply:
I know it's just rewrote, but I'm confused about those parameters in URL: `service_name=HDFS&service_config_version=1`

Reply:
It's a rest API provided by Ambari. You can get more details from https://docs.microsoft.com/en-us/azure/hdinsight/hdinsight-hadoop-manage-ambari-rest-api.

Reply:
There is a notice in that page:

> Important
These examples return the first configuration applied to the server (service_config_version=1) which contains this information. If you retrieve a value that has been modified after cluster creation, you may need to list the configuration versions and retrieve the latest one.

I would suggest to add a `String serviceConfigVersion` parameter in the funcation.

Reply:
Fixed. Thanks.